org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalReplacementProvider.java

```diff
@@ -239,10 +249,15 @@ private void appendGuessingCompletion(StringBuilder buffer, CompletionProposal p
 			}
 
 			char[] argument = parameterNames[i];
-
-			buffer.append("{{");
+			if(supportSnippets){
+				buffer.append("${");
+				buffer.append(Integer.toString(i+1));
+				buffer.append(":");
+			}
 			buffer.append(argument);
```

Comment:
Unfortunately, there is no support for signature help in Vim, so YouCompleteMe does not have signature help for any supported language.

Incidentally, ycmd already de-duplicates method suggestions in our C++ completer, so we could easily handle that in the client. Right now, inserting text which the user just has to manually delete means it isn't usable in our model. I hacked the code to just not supply any of the `( .... );` and it works lovely. 

I realise you have multiple clients to think about, but i'd be curious in what environment completing text that the user has to just remove would be all that useful ^_^ (in a nice way of course!)